packages/components/src/suggestions/item.tsx

```diff
@@ -0,0 +1,87 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import type { ForwardRefRenderFunction, FocusEvent, MouseEvent } from 'react';
+import classNames from 'classnames';
+import { escapeRegExp } from 'lodash';
+
+function escapeRegExpWithSpace( str: string ) {
+	return escapeRegExp( str ).replace( /\s/g, '\\s' );
+}
+
+const createTextWithHighlight = ( text: string, query: string ) => {
+	const re = new RegExp( '(' + escapeRegExpWithSpace( query ) + ')', 'gi' );
+	const parts = text.split( re );
+
+	// Replaces char code 160 (&nbsp;) with 32 (space)
+	const match = query.toLowerCase().replace( /\s/g, ' ' );
+
+	return parts.map( ( part, i ) => {
+		const key = text + i;
+		const lowercasePart = part.toLowerCase();
+		const spanClass = classNames( 'suggestions__label', {
+			'is-emphasized': lowercasePart === match,
+		} );
+
+		return (
+			<span key={ key } className={ spanClass }>
+				{ part }
+			</span>
+		);
+	} );
+};
+
+type Props = {
+	label: string;
+	hasHighlight: boolean;
+	query: string;
+	onMount: () => void;
+	onMouseDown: () => void;
+	onMouseOver: () => void;
+};
+
+const InnerItem: ForwardRefRenderFunction< HTMLButtonElement, Props > = (
+	{ label, hasHighlight, query, onMount, onMouseDown, onMouseOver },
+	forwardedRef
+) => {
+	React.useEffect( () => {
+		onMount();
+	}, [ onMount ] );
+
+	const handleMouseDown = React.useCallback(
+		( event: MouseEvent | FocusEvent ) => {
+			event.stopPropagation();
+			event.preventDefault();
+			onMouseDown();
+		},
+		[ onMouseDown ]
+	);
+
+	const handleMouseOver = React.useCallback( () => {
```

Comment:
Is `useCallback` really giving us much extra performance in these two cases, or is it just added overhead?

In this example, it seems like it might be ok to just use `onMouseOver` directly since we don't need to do anything extra with it. E.g. we are just re-computing the cache when the reference changes, which seems  redundant.

I'm no expert on `useCallback` though :)

Reply:
I think you're correct because we're passing it to a regular intrinsic element anyway.